src/XRP/model/xrp-payment.ts

```diff
@@ -60,14 +58,29 @@ export default class XrpPayment {
       payment.getPathsList()?.length > 0
         ? payment.getPathsList().map((path) => XrpPath.from(path))
         : undefined
+    if (paths && amount.drops) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Payment protobuf `paths` field should be omitted for XRP-to-XRP interactions.',
+      )
+    }
 
-    // If the sendMax field is set, it must be able to be transformed into an XrpCurrencyAmount.
     const paymentSendMaxValue = payment.getSendMax()?.getValue()
     const sendMax =
       paymentSendMaxValue && XrpCurrencyAmount.from(paymentSendMaxValue)
-    if (paymentSendMaxValue && !sendMax) {
-      return undefined
+    if (sendMax && amount.drops) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Payment protobuf `sendMax` field should be omitted for XRP-to-XRP interactions.',
+      )
     }
+    // For non-XRP amounts, the nested field names in `sendMax` MUST be lower-case.
+    // `sendMax` must be supplied for cross-currency/cross-issue payments.
```

Comment:
I believe you can implement this check in much the same way you did the `path` and XRP-to-XRP payments... I think we'd just ask about `amount.issuedCurrency` or whatever the equivalent is, && sendMax.  I think that any payment that's not XRP-to-XRP requires a sendMax... pretty sure about that.  We can always investigate and change later if need be.